EventsExpress/ClientApp/src/containers/inventory-item.js

```diff
@@ -24,9 +24,21 @@ class InventoryItemWrapper extends Component {
         this.onWillTake = this.onWillTake.bind(this);
     }
 
+    shouldComponentUpdate(nextProps, nextState) {
+        if (this.props === nextProps && this.state === nextState)
+            return false
+        if (this.props.usersInventories.isPending !== nextProps.usersInventories.isPending) {
+            const { item, user } = nextProps
+            if (!this.state.isWillTake && nextProps.usersInventories.data.some(e => e.userId === user.id && e.inventoryId === item.id))
+                this.onAlreadyGet()
```

Comment:
You are mutating the state based on the props that are coming in. Should it not go into the `componentDidUpdate` instead?